libs/ui-watt/src/lib/components/toast/watt-toast.component.ts

```diff
@@ -0,0 +1,137 @@
+/**
+ * @license
+ * Copyright 2020 Energinet DataHub A/S
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License2");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { CommonModule } from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  HostBinding,
+  Inject,
+  Optional,
+  ViewEncapsulation,
+} from '@angular/core';
+import {
+  MatSnackBarModule,
+  MatSnackBarRef,
+  MAT_SNACK_BAR_DATA,
+} from '@angular/material/snack-bar';
+import { fromEvent, repeat, Subscription, takeUntil, tap, timer } from 'rxjs';
+
+import { WattIconModule } from '../../foundations/icon';
+import { WattButtonModule } from '../button';
+import { WattSpinnerModule } from '../spinner';
+
+export type WattToastType =
+  | 'success'
+  | 'info'
+  | 'warning'
+  | 'danger'
+  | 'loading';
+
+export interface WattToastConfig {
+  duration?: number;
+  type?: WattToastType;
+  message: string;
+  action?: (wattToastRef: WattToastRef) => void;
+  actionLabel?: string;
+}
+
+export type WattToastRef = MatSnackBarRef<WattToastComponent>;
+
+/**
+ * Usage:
+ * `import { WattToastComponent } from '@energinet-datahub/watt';`
+ */
+@Component({
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+  selector: 'watt-toast',
+  styleUrls: ['./watt-toast.component.scss'],
+  templateUrl: './watt-toast.component.html',
+  standalone: true,
+  imports: [
+    CommonModule,
+    MatSnackBarModule,
+    WattButtonModule,
+    WattIconModule,
+    WattSpinnerModule,
+  ],
```

Comment:
praise: nice changes 🌟

question: why do we need to expose the `WattToastComponent` from Watt when it's only used in the `WattToastService`? The only valid use case is for testing. Other than that, are other developers actually going to write `watt-toast` in their templates?

suggestion: let's keep the `WattToastComponent` internal to Watt. We still need to expose `WattToastService` and `WattToastModule` but the only thing the module would do is import `MatSnackBarModule` because that's the only thing needed by the Material Toast component in order to work properly. The `forRoot` pattern will be important in this case because the `WattToastModule` is going to be imported only once in `DhCoreShellModule`. It shouldn't be possible to import `WattToastModule` in other modules. So other developers will only need to import `WattToastService` in order to open a toast. Does this make sense?

Reply:
Should be resolved now :)